pil_analyzer/src/pil_analyzer.rs

```diff
@@ -499,7 +479,39 @@ impl<T: FieldElement> PILAnalyzer<T> {
         }
     }
 
-    fn process_selected_expression(
+    fn evaluate_expression(&self, expr: ::ast::parsed::Expression<T>) -> Result<T, String> {
+        Evaluator {
+            constants: &self.constants,
+            definitions: &self.definitions,
+            function_cache: &Default::default(),
+            variables: &[],
+        }
+        .evaluate(&self.process_expression(expr))
+    }
+
+    fn process_expression(&self, expr: ::ast::parsed::Expression<T>) -> Expression<T> {
+        ExpressionProcessor::new(self).process_expression(expr)
+    }
+}
+
+/// The ExpressionProcessor turns parsed expressions into analyzed expressions.
+/// It main job is to resolve references:
+/// It turns simple references into fully namespaced references and resolves local function variables.
+/// It also evaluates expressions that are required to be compile-time constant.
```

Comment:
Shouldn't it evaluate constants then?

Reply:
I mean it evaluates expressions. If that includes calling some other functions or evaluating constants, that will be done.